### DISCUSSION TITLE
Ignoring Vulnerabilities for a Period of Time

### DISCUSSION THREAD

**Discussion by muks198** [aquasecurity/trivy#2991] [Q&A]
Hi,

New to REGO and attempting to construct a policy that will ignore a specific vulnerability for a period of time using the --ignore-policy command.

The following works to ignore a vulnerability in general:
```
ignore {
	input.VulnerabilityID == "CVE-XYZ"

}
```

How would I go about creating a rule that would resurface the vulnerability after a specified date?

Thanks in advance!
M.

**Comment by ctrung** [comment]
Hi, 

I'd like to do the same thing for all new vulnerabilities based on their publishing date and a given number of whatever unit of time.  \
The idea is to give some leeway to the dev teams to updates their dependencies. Something like `trivy fs --ignore-new-vuln 14` to ignore new vulnerabilities for 14 days would do the job. There should be a way to forbid this behaviour for a defined set of vulnerabilities as well (eg. log4shell), so a more elegant way of doing this might be to extend the .trivyignore file ? 

```
$ cat .trivyignore
# Accept the risk for new ones for 14 days but nor for log4shell and spring4shell
* new-for:14 exclude:CVE-2021-44228,CVE-2022-22965
```


For your particular usecase, you might have a look at Trivy >= 0.32 that supports the exp attribute in a .trivyignore file. 
```
$ cat .trivyignore
# Accept the risk until 2023-01-01
CVE-2019-14697 exp:2023-01-01
```
Source : https://aquasecurity.github.io/trivy/v0.34/docs/vulnerability/examples/filter/#by-vulnerability-ids

**Comment by ricardbejarano** [comment]
I would also greatly enjoy this.

cc @nikpivkin @knqyf263 @DmitriyLewen or @simar7 maybe? going off GitHub contribution stats...

**Comment by itaysk** [comment]
you can see such ignore rule with rego here: https://github.com/itaysk/cnscna24-scan-patch-vex/blob/main/trivyignore.rego#L14
if you're interested, this is part of a a talk: https://youtu.be/Tu9qPe1cjuQ
